test/rules/experiments/nnbd/rules/prefer_constructors_over_static_methods.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N prefer_constructors_over_static_methods`
+
+class A {
+  static final array = <A>[];
+
+  A.internal();
+
+  static A bad1() => // LINT
+      new A.internal();
+
+  static A get newA => // LINT
+      new A.internal();
+
+  static A bad2() { // LINT
+    final a = new A.internal();
+    return a;
+  }
+
+  static A good1(int i) { // OK
```

Comment:
I'm not sure why this one is OK; it seems like it could easily be a factory constructor:

```
factory A(int i) => array[i];
```

On the other hand, given a static field:

```
static final Map<String, A> = {};
```

the following method can't be converted because it can return `null`:

```
static A bad3(String key) {
  return map[key];
}
```

Reply:
👍